### ISSUE TITLE
Flakes in the E2E Conformance tests labeled as Serial

### ISSUE THREAD

**Issue by fabriziopandini** [kubernetes-sigs/cluster-api#4077]
**What steps did you take and what happened:**
Recently we are observing an increasing number of failures in E2E conformance tests labeled as [Serial] e.g.
- Kubernetes e2e suite.[sig-apps] Daemon set [Serial] should run and stop complex daemon [Conformance]
- Kubernetes e2e suite.[sig-scheduling] SchedulerPreemption [Serial] validates basic preemption works [Conformance]
- Kubernetes e2e suite.[sig-scheduling] SchedulerPreemption [Serial] validates lower priority pod preemption by critical pod [Conformance]

This is due to the fact that in our E2E test framework we are running tests in parallel 5 ginkgo nodes.

So now we have two options:
- Remove parallelism; this will increase the overall test duration to ~1 for the conformance test only.
- Skip test labeled as [Serial] 

I would like to hear other opinions as well before proceeding with the change.

Personally, I'm +1 to choose option 2, given that:
- CAPD on prow (docker in docker in docker) is fast, but also easily subject to noisy neighbors, so I prefer stability over depth of our conformance signal.
- We will get a better signal about conformance from providers.
- this is the same approach followed by kubeadm and the CI signal there is almost stable there

@vincepri @CecileRobertMichon @randomvariable @detiber opinions?

**Environment:**

- Cluster-api version: main

/kind bug
/area testing

**Comment by vincepri** [comment]
+1 to skip the serial tests, although would like to hear from others that have more experience in this area

**Comment by neolit123** [comment]
we are skipping these in kubeadm tests.
serial tests are meant to run in a separate less-freq job.

**Comment by fabriziopandini** [comment]
/help
/good-first-issue

To fix this add `ginkgo.skip: \[Serial\]` to https://github.com/kubernetes-sigs/cluster-api/blob/master/test/e2e/data/kubetest/conformance.yaml, send a PR, and then check the output of conformance tests after triggering `/test pull-cluster-api-e2e-full-main`; no tests labeled [Serial] should be there for the latest run...

**Comment by furkatgofurov7** [comment]
@fabriziopandini I can test it locally and send a patch with suggested changes. 

/assign
